stream-ingest/src/main/java/com/dellemc/oe/ingest/EventWithTimestampWriter.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package com.dellemc.oe.ingest;
+
+import com.dellemc.oe.serialization.JsonNodeSerializer;
+import com.dellemc.oe.util.CommonParams;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.EventStreamClientFactory;
+import io.pravega.client.admin.StreamManager;
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.impl.DefaultCredentials;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+/**
+ * A simple example app that uses a Pravega Writer to write to a given scope and stream.
+ */
+public class EventWithTimestampWriter {
+    // Logger initialization
+    private static final Logger LOG = LoggerFactory.getLogger(JSONWriter.class);
+
+    public final String scope;
+    public final String streamName;
+    public final URI controllerURI;
+
+    public EventWithTimestampWriter(String scope, String streamName, URI controllerURI) {
+        this.scope = scope;
+        this.streamName = streamName;
+        this.controllerURI = controllerURI;
+    }
+
+    // Create a JSON data for testing purpose
+    public static ObjectNode createJSONData() {
+        ObjectNode message = null;
+        try {
+            String data = "{\"sensorid\":" + Math.random() + ",\"time\":" + System.currentTimeMillis() + ",\"value\":" + Math.random();
+            // Deserialize the JSON message.
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(data);
+            message = (ObjectNode) jsonNode;
+            LOG.info("@@@@@@@@@@@@@ DATA >>>  " + message.toString());
+            return message;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void main(String[] args) {
+        final String scope = CommonParams.getScope();
+        final String streamName = CommonParams.getStreamName();
+        final String routingKey = CommonParams.getRoutingKeyAttributeName();
+        final URI controllerURI = CommonParams.getControllerURI();
+        JSONWriter ew = new JSONWriter(scope, streamName, controllerURI);
+        ew.run(routingKey);
+    }
+
+    public void run(String routingKey) {
```

Comment:
getRoutingKeyAttributeName should be the *key* of the routing key which would be "sensorid" here. The actual routing key needs to be retrieved from the JSON document as the value of this the "sensorid" key. On this line, change routingKey to routingKeyAttributeName.